catt/youtube.py

```diff
@@ -0,0 +1,381 @@
+"""
+Controller to interface with the YouTube-app.
+"""
+
+import re
+import threading
+from json import JSONDecodeError
+
+import requests
+
+from pychromecast.controllers import BaseController
+
+YOUTUBE_BASE_URL = "https://www.youtube.com/"
+YOUTUBE_WATCH_VIDEO_URL = YOUTUBE_BASE_URL + "watch?v="
+
+# id param is const(YouTube sets it as random xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx so it should be fine).
+RANDOM_ID = "12345678-9ABC-4DEF-0123-0123456789AB"
+VIDEO_ID_PARAM = '%7B%22videoId%22%3A%22{video_id}%22%2C%22currentTime%22%3A5%2C%22currentIndex%22%3A0%7D'
+TERMINATE_PARAM = "terminate"
+
+REQUEST_URL_SET_PLAYLIST = YOUTUBE_BASE_URL + "api/lounge/bc/bind?"
+BASE_REQUEST_PARAMS = {"device": "REMOTE_CONTROL", "id": RANDOM_ID, "name": "Desktop&app=youtube-desktop",
+                       "mdx-version": 3, "loungeIdToken": None, "VER": 8, "v": 2, "t": 1, "ui": 1, "RID": 75956,
+                       "CVER": 1}
+
+SET_PLAYLIST_METHOD = {"method": "setPlaylist", "params": VIDEO_ID_PARAM, "TYPE": None}
+REQUEST_PARAMS_SET_PLAYLIST = {**BASE_REQUEST_PARAMS, **SET_PLAYLIST_METHOD}
```

Comment:
This syntax was introduced in python3.5, making ci-builds fail on older pythons. I'm unsure how to fix this.